test/evergreen_develop_only_variants.yml

```diff
@@ -0,0 +1,122 @@
+# Build variants that need to be run only on the develop branch.
```

Comment:
@lukech could we move the contents of this file to `test/evergreen_develop.yml`? 
Seems a bit unnecessary having an extra file.

Reply:
Updated in [41617f2](https://github.com/wiredtiger/wiredtiger/pull/8587/commits/41617f2ae9f289fbc1e155236b0c4602ae43264a)